File: LabProject/src/main/java/agh/ics/oop/MapDirection.java
package agh.ics.oop;

public enum MapDirection {
    NORTH,
    SOUTH,
    WEST,
    EAST;

    @Override
    public String toString() {
        String text=switch(this){
            case NORTH -> "Północ";
            case SOUTH -> "Południe";
            case WEST -> "Zachód";
            case EAST -> "Wschód";
            default -> null;
        };
        return text;
    }

    public MapDirection next(){
        MapDirection next_res = switch(this){
            case NORTH -> EAST;
            case SOUTH -> WEST;
            case WEST -> NORTH;
            case EAST -> SOUTH;
            default -> null;
        };
        return next_res;
    }

    public MapDirection previous(){
        MapDirection prev_res= switch(this){
            case NORTH -> WEST;
            case SOUTH -> EAST;
            case WEST -> SOUTH;
            case EAST -> NORTH;
            default -> null;
        };
        return prev_res;
    }

    public Vector2d toUnitVector(){
        Vector2d u=switch(this){
            case NORTH -> new Vector2d(0,1);
            case SOUTH -> new Vector2d(0,-1);
            case WEST -> new Vector2d(-1,0);
            case EAST -> new Vector2d(1,0);
            default -> null;
        };
        return u;
    }
}

File: KotlinLab/src/main/kotlin/MapDirection.kt


enum class MapDirection {
    NORTH, SOUTH, WEST, EAST;


    override fun toString(): String = when(this){
            NORTH -> "Północ"
            SOUTH -> "Południe"
            WEST -> "Zachód"
            EAST -> "Wschód"
        }

    fun next(): MapDirection = when(this){
        NORTH -> EAST;
        SOUTH -> WEST;
        WEST -> NORTH;
        EAST -> SOUTH;
    }
    fun previous(): MapDirection = when(this){
        NORTH -> WEST;
        SOUTH -> EAST;
        WEST -> SOUTH;
        EAST -> NORTH;
    }

}